qcodes/instrument/meta/meta_parameter.py

```diff
@@ -0,0 +1,174 @@
+        self._namedtuple = namedtuple(name, parameter_names)
+
+    def get_parameters(self) -> Any:
+        return self._namedtuple(*[_p.get() for _p in self.parameters.values()])
+
+    def _set_from_dict(self, calling_dict: Dict[str, ParamRawDataType]) -> None:
+        for name, p in list(self.parameters.items()):
+            p.set(calling_dict[name])
+
+
+class MetaParameter(_BaseParameter):
+    """
+    Meta parameter that returns one or more endpoint parameter values
+    
+    Args:
+        name: Meta parameter name
+        endpoints: One or more endpoint parameters to alias
+        endpoint_names: One or more endpoint names
+        unit: The unit of measure. Use ``''`` for unitless.
+        setter: Optional setter function to override
+            endpoint parameter setter. Defaults to None.
+    """
+    def __init__(
+        self,
+        name: str,
+        endpoints: Tuple[Parameter],
+        endpoint_names: Tuple[str],
+        unit: str = None,
+        setter: callable = None,
+        **kwargs
+    ):
+        super().__init__(name, **kwargs)
+        self._endpoints = endpoints
+        self._setter = setter
+        self.unit = unit if unit is not None else ''
+        endpoint_names = endpoint_names or [_e.name for _e in endpoints]
+        self._group = None
+
+        # Generate sub parameters if >1 endpoints,
+        # e.g. my_meta_param.X, my_meta_param.Y
+        if len(endpoints) > 1:
+            self._parameters = self._sub_parameters(endpoints, endpoint_names)
+        self._set = self.set
+        self.set = self._set_override
+
+    @property
+    def group(self) -> Optional['MetaGroup']:
+        """
+        The group that this parameter belongs to.
+        """
+        return self._group
+
+    def _sub_parameters(self, endpoints: Tuple[Parameter], endpoint_names: Tuple[str]):
+        """Generate sub parameters if there are > 1 endpoints."""
+        parameters = []
+        for endpoint, endpoint_name in zip(
+            endpoints,
+            endpoint_names
+        ):
+            parameter = self.add_parameter(
+                cls=self.__class__,
+                name=endpoint_name,
+                endpoint=endpoint
+            )
+            parameters.append(parameter)
+        return tuple(parameters)
+
+    def add_parameter(
+        self,
+        cls,
+        name: str,
+        endpoint: Parameter
+    ) -> Parameter:
+        """Add a sub parameter to the meta parameter
+
+        Args:
+            name: Sub parameter name
+            endpoint: Endpoint parameter object
+
+        Returns:
+            Parameter: Parameter instance
+        """
+        assert not hasattr(
+            self, name
+        ), f"Duplicate parameter name {name}."
+        parameter = cls(
+            name=name,
+            endpoints=(endpoint,),
+            endpoint_names=(endpoint.name,),
+            instrument=None
+        )
+        setattr(self, name, parameter)
+        return parameter
```

Comment:
Alternatively this should use the Metadatable interface to register parameters

Reply:
This happens since Parameter was never intended to have sub components in the first place. I unfortunately think that using parameters for this purpose will result in more issues down the line but we can try it out now and see what works out

Reply:
That is a fair point, I think it would be more consistent to create a new parameter called e.g. `lockin_frequency` rather than `lockin.frequency`.